pandas/io/parsers.py

```diff
@@ -2755,23 +2755,24 @@ def _check_for_bom(self, first_row):
         if first_elt != _BOM:
             return first_row
 
-        first_row = first_row[0]
+        first_row_bom = first_row[0]
```

Comment:
Hmm so the assumption here is that if there is a BOM the length of the first parsed element will be greater than 1, correct? Not sure if that's right or wrong just want to verify

Reply:
Yes, there is an assumption that the first element will have a BOM, I do not know whether it is correct or not. But initially the error was in the other. We got the list. Mutated it into one element. Cleared this item from the BOM and given that this element is wrapped in a list.